helper-cli/src/funTest/assets/package-list.yml

```diff
@@ -0,0 +1,23 @@
+---
+name: "Example project name"
+vcs_type: "Git"
+vcs_url: "https://github.com/example/project.git"
+vcs_revision: "2222222222222222222222222222222222222222"
+vcs_path: "vcs-path/project"
+dependencies:
+  - id: "NPM::example-dependency-one:1.0.0"
+    vcs_type: "Git"
+    vcs_url: "https://github.com/example/depedency-one.git"
+    vcs_revision: "0000000000000000000000000000000000000000"
+    vcs_path: "vcs-path/dependency-one"
+    source_artifact_url: "https://example.org/example-dependency-one.zip"
```

Comment:
Have you read the commit message, which has some rationale why I decided against that?

Reply:
Ok, so can we agree on the exact format before I change the implementation @sschuberth  @mnonnenmacher ?
I want to avoid further unnecessary ping pong.

1. For VCS i cannot use VcsInfo class because revision is mandatory which is undesired.
2. For Source artifact URL do you require me to use the RemoteArtifact class which then also requires specifying the Hash algorithm?

I prefer to keep it as-is, but as you seem to have a concrete vision how this needs to change - it'd be great if we could talk as specific as possible.

Reply:
> For VCS i cannot use VcsInfo class because revision is mandatory which is undesired.

Could you please explain *why* it is undesired (for your use cases)? And what are the semantics if the revision is omitted, i.e. what revision would get checked out by the scanner? I'm asking to maybe find a viable alternative to omitting the revision.

Reply:
If the revision is omit, then ORT would match the version name against git tags.
So, if the file is generated, the user can choose between

1. Using the tag matching mechanism and then fetching subsequent issues via Vcs revision curations
2. Making the generation output the revision

I believe it's valuable to have both options available to the user, which requires the revision to be optional.

Reply:
> I believe it's valuable to have both options available to the user

I basically agree.

> which requires the revision to be optional.

How about requiring to explicit set the revision to an empty string to trigger case 1? I believe we handle that case already gracefully:

https://github.com/oss-review-toolkit/ort/blob/97acc35f3fe343b1b79cb9265b51019ecd8ae63e/downloader/src/main/kotlin/VersionControlSystem.kt#L327

Reply:
Hm, if something is not needed I think it's more natural to assign a default value, but I can live with that.

Maybe, let's regard one alternative: Would it work out to add a default value for `VcsInfo.revision` so that we can just use the `VcsInfo` class.

From your comment  @sschuberth , I interpret that you assume that I should use the VcsInfo class as well as the RemoteArtifact class. Is that correct ?

Reply:
> Would it work out to add a default value for VcsInfo.revision so that we can just use the VcsInfo class.

Maybe yes, but I'd like to know also what @mnonnenmacher thinks.

To throw in another technically possible alternative (just to mention it, not saying that it's a good approach): This command could use a custom deserializer for `VcsInfo` to apply a default revision only in this context.

> I should use the VcsInfo class as well as the RemoteArtifact class. Is that correct ?

I'd prefer to make as much reuse of your existing model classes as possible, yes.

Reply:
> I'd prefer to make as much reuse of your existing model classes as possible, yes.

Is this "yes" an answer to my question, or limited to "as much as possible" whereas "possible" is a matter of opinion?

Reply:
It's a "yes" as the answer to your question. I realized too late that I shouldn't have written "possible" as that leaves too much room for interpretation again.

Reply:
But well, I believe we're simply not through with the discussion yet. My main point is to get more structure by nesting, and that's why reusing the existing model classes seems natural. However, if you also want different *semantics* than those implied by the existing model classes, it *might* indeed make sense to come up with custom model classes in the scope of this command. I'd probably easier if we could discuss this by voice, but @mnonnenmacher does not seem to be available today, and I'm not on Monday at the Kotlin Dev meeting.

Reply:
Makes sense to discuss this by voice, I'll try to arrange a date.

Reply:
I would actually prefer not to reuse the existing model classes for two reasons:
* If kotlinx.serialization is used as I requested in another comment, the existing model classes cannot easily be used.
* Using dedicated classes for the command provides better decoupling, so changes in ORT model classes cannot break the deserialization for the command, just the mapping code needs to be updated.

Reply:
Using dedicated classes now.